L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,301 @@
+    HGCalTriggerBackendDetId tc_fpga(fpgaId);
+    if (!(tc_fpga.isStage1FPGA()) || (tc_fpga.sector() != 0) || (tc_fpga.zside() < 0))
+      continue;
+    if (tc_fpga.label() != the_fpga_)
+      continue;
+
+    // retrieve information to be saved
+    int triggerCellSubdet = 0;
+    int triggerCellLayer = 0;
+    int triggerCellUEta = 0;
+    int triggerCellVPhi = 0;
+
+    bool isScintillatorCell = triggerTools_.isScintillator(id);
+    if (isScintillatorCell) {
+      HGCScintillatorDetId id_sc(triggercell);
+      triggerCellSubdet = id_sc.subdet();
+      triggerCellLayer = id_sc.layer();
+      triggerCellUEta = id_sc.ietaAbs();
+      triggerCellVPhi = id_sc.iphi();
+    } else {
+      HGCalTriggerDetId id_si_trig(triggercell);
+      triggerCellSubdet = id_si_trig.subdet();
+      triggerCellLayer = id_si_trig.layer();
+      triggerCellUEta = id_si_trig.triggerCellU();
+      triggerCellVPhi = id_si_trig.triggerCellV();
+    }
+    GlobalPoint position = triggerGeometry_->getTriggerCellPosition(triggercell);
+    float triggerCellX = position.x();
+    float triggerCellY = position.y();
+    float triggerCellZ = position.z();
+    float triggerCellEta = position.eta();
+    float triggerCellPhi = position.phi();
+    float triggerCellRoverZ = sqrt(triggerCellX * triggerCellX + triggerCellY * triggerCellY) / triggerCellZ;
+
+    uint32_t triggerCellRoZBin = theAlgo_.rozBin(triggerCellRoverZ, roz_min_, roz_max_, roz_bins_);
+    double triggerCellRotatedPhi = theAlgo_.rotatedphi(triggerCellX, triggerCellY, triggerCellZ, tc_fpga.sector());
+    uint32_t triggerCellPhiBin = theAlgo_.phiBin(triggerCellRoZBin, triggerCellRotatedPhi, phi_edges_);
+
+    uint32_t tcAddress = getTCaddress(triggerCellUEta, triggerCellVPhi, isScintillatorCell);
+
+    // save TC info into JSON
+    json theTC;
+    theTC["tcid"] = tcAddress;
+    theTC["subdet"] = triggerCellSubdet;
+    theTC["layer"] = triggerCellLayer;
+    theTC["ueta"] = triggerCellUEta;
+    theTC["vphi"] = triggerCellVPhi;
+    theTC["x"] = triggerCellX;
+    theTC["y"] = triggerCellY;
+    theTC["z"] = triggerCellZ;
+    theTC["roz"] = triggerCellRoverZ;
+    theTC["eta"] = triggerCellEta;
+    theTC["phi"] = triggerCellPhi;
+    theTC["roz_bin"] = triggerCellRoZBin;
+    theTC["phi_bin"] = triggerCellPhiBin;
+
+    std::string strModId = std::to_string(moduleHash);
+    tmp_json[strModId].push_back(theTC);
+  }
+
+  // fill output JSON
+  for (auto& module_json : tmp_json.items()) {
+    json j1;
+    j1["hash"] = module_json.key();
+    j1["tcs"] = module_json.value();
+    json_file["TriggerCellMap"]["Module"].push_back(j1);
+  }
+}
+
+uint32_t HGCalBackendStage1ParameterExtractor::getTCaddress(int& tc_ueta, int& tc_vphi, bool isScintillator) {
+  // transform HGCalHSc TC coordinates to define a TC address in [0,47]
+  if (isScintillator) {  // HGCalHSc
+    tc_vphi = (tc_vphi - 1) % 4;
+    if (tc_ueta <= 3) {
+      tc_ueta = tc_ueta;
```

Comment:
You likely don't mean `tc_ueta = tc_ueta` here. This is what is causing the build errors. What should this actually be?